pkg/distributor/otel.go

```diff
@@ -71,169 +72,165 @@ func OTLPHandler(
 ) http.Handler {
 	discardedDueToOtelParseError := validation.DiscardedSamplesCounter(reg, otelParseError)
 
-	return otlpHandler(maxRecvMsgSize, requestBufferPool, sourceIPs, retryCfg, push, logger, func(ctx context.Context, r *http.Request, maxRecvMsgSize int, buffers *util.RequestBuffers, req *mimirpb.PreallocWriteRequest, logger log.Logger) error {
-		contentType := r.Header.Get("Content-Type")
-		contentEncoding := r.Header.Get("Content-Encoding")
-		var compression util.CompressionType
-		switch contentEncoding {
-		case "gzip":
-			compression = util.Gzip
-		case "lz4":
-			compression = util.Lz4
-		case "":
-			compression = util.NoCompression
-		default:
-			return httpgrpc.Errorf(http.StatusUnsupportedMediaType, "unsupported compression: %s. Only \"gzip\", \"lz4\", or no compression supported", contentEncoding)
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
A note for the reviewer: this chunk looks "weird" in the github's diff. All it does is
1. inlines the `otlpHandler` wrapper function;
2. introduces the `otlpConverter`, that holds the paring errors.

The rest of the logic (in this chunk) is the same as it was before.

Reply:
The diff is understandable in itself, but why do you need to inline `otlpHandler`? I don't like that change, because `OTLPHandler` becomes a mega function. Usually you will go the other way around, and break down such mega functions so they're more readable.